### ISSUE TITLE
Add Netlify previews

### ISSUE THREAD

**Issue by techtonik** [badges/shields#960]
https://www.netlify.com/ can greatly speed up website previews if integrated.

**Comment by paulmelnikow** [comment]
Hi, thanks for the suggestion. I'm not sure I understand how you're suggesting this project use it. Could you elaborate?

**Comment by techtonik** [comment]
@paulmelnikow see Netlify check results here - https://github.com/kubernetes/kubernetes.github.io/pull/3522

**Comment by paulmelnikow** [comment]
I see, there's a Github integration which generates a built copy of the website for a PR.

An alternative we could consider is [Heroku Review Apps](https://devcenter.heroku.com/articles/github-integration-review-apps), which puts up a live version of the app against a PR. It would let reviewers and commenters interactively test example badges without having to install and run the review branch locally. We could paste live badge links into the PR too.

**Comment by Daniel15** [comment]
Netlify is a host for static websites. Their service wouldn't work for this project, as we need to serve badges dynamically. 

However, something like Heroku Review Apps is a great idea! Either that or spin up a Docker container on our own VPS. I'm not sure how the shields.io site is currently hosted, but the approach that is chosen should be close to whatever is done in production so it's an accurate representation of how the change would behave 😃

**Comment by techtonik** [comment]
@Daniel15 shields is a static site. no? There is no backend logic IIRC.

**Comment by Daniel15** [comment]
@techtonik - It's not static, otherwise the badges wouldn't actually work as expected. The badges need to load data in real-time, for example:
![](https://img.shields.io/github/downloads/atom/atom/total.svg)

In order to be static, every single badge would need to be computed ahead of time rather than dynamically generated at runtime. This is impossible as there's a practically infinite number of possible combinations :)

**Comment by techtonik** [comment]
Server is needed, indeed. Heroku Review App looks neat, but from what I could see it is a paid option and I don't understand the 5 days lifespan. Netlify static copy could be available indefinitely.

**Comment by paulmelnikow** [comment]
A snapshot of the index page could be useful for some PRs as a basic confidence check. Though if we have a review apps running while PRs are open, we probably wouldn't need the static copy too.

I'm not sure review apps would end up costing anything. If they did we could ask Heroku if they'd sponsor us.

Spinning up our own docker containers is an interesting idea. I wonder if there is a tool that would integrate with github and do most of the work for us. It would be a fun tool to build – though it would be a project of its own!

Test–prod parity is a good principle to strive for, though we should balance that against how much work it is to build and maintain our own automation system. 

Tangentially related: something I'd really like to see is a scalability test: a test suite which simulates real traffic against a test server. We could run it before a deploy, on PRs which seem risky, maybe weekly on master. That's the area where test–prod parity would be _really_ important.

**Comment by techtonik** [comment]
So are PR previews functional?

**Comment by techtonik** [comment]
> Spinning up our own docker containers is an interesting idea. I wonder if there is a tool that would integrate with github and do most of the work for us. It would be a fun tool to build – though it would be a project of its own!

There are tools like https://github.com/drone/drone but for GitHub integration they need to have some authenticated point of access to submit results.

> Tangentially related: something I'd really like to see is a scalability test: a test suite which simulates real traffic against a test server. We could run it before a deploy, on PRs which seem risky, maybe weekly on master. That's the area where test–prod parity would be really important.

Is it already possible to catch anonymized traffic from current site? Or at least get some statistics?

**Comment by paulmelnikow** [comment]
Heroku Review Apps are live for PRs. They need to be triggered by one of the maintainers, but it's an easy thing to do when it's useful.

Stats, unfortunately not, though this would definitely be useful! That's in #966.

**Comment by techtonik** [comment]
Is there a documentation about how triggering process works?

**Comment by paulmelnikow** [comment]
Yes, our maintainer documentation could be better. There is so much to do in this project! I'd be happy for you to add it to `CONTRIBUTING.md`.

Alternatively you could work on automating it. By design, Heroku doesn't auto deploy review apps from forks. But maybe we could configure probot to respond to a maintainer comment and deploy from that, chat-ops style.

**Comment by techtonik** [comment]
Well, I am not a maintainer, so I can not write about something I can not try. Only after there is a description of the manual procedure, it can be automated. I guess right now it is like this - somebody with Heroku privileges (who?) goes to Heroku, chooses (or enter?) the PR and triggers a build for this PR. I mean I would be happy to read more and infer the correct order operation if not this:
![image](https://user-images.githubusercontent.com/515889/43150389-63001d82-8f9c-11e8-9117-7c428caa1c7d.png)
=)
